{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import bs4\n",
    "from bs4 import BeautifulSoup as soup\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HTML Parsing\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Print List of Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.html\n",
      "10.html\n",
      "100.html\n",
      "1000.html\n",
      "10000.html\n"
     ]
    }
   ],
   "source": [
    "collection = \"html\"\n",
    "\n",
    "n_test=5\n",
    "file_list = os.listdir(collection)\n",
    "# for i in range(len(file_list)):\n",
    "for i in range(n_test):\n",
    "    print(file_list[i])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Sample Test File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.html\n"
     ]
    }
   ],
   "source": [
    "test_file = file_list[1]\n",
    "print(test_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load HTML File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "<div class=\"content_part hid\" id=\"article_participants\">\n",
      "<p>\n",
      "                  Cancer Genetics, Inc. (NASDAQ:\n",
      "                  <a href=\"http://seekingalpha.com/symbol/cgix\" title=\"Cancer Genetics, Inc.\">\n",
      "                    CGIX\n",
      "                  </a>\n",
      "                  )\n",
      "                </p>\n",
      "<p>\n",
      "                  Q1 2015 Earnings Conference Call\n",
      "                </p>\n",
      "<p>\n",
      "                  May 12, 2015 08:30 AM ET\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Executives\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Brian - IR\n",
      "                </p>\n",
      "<p>\n",
      "                  Panna Sharma - President and CEO\n",
      "                </p>\n",
      "<p>\n",
      "                  Edward Sitar - CFO and Treasurer\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Analysts\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Paul Knight - Janney Montgomery Scott\n",
      "                </p>\n",
      "<p>\n",
      "                  Ben Haynor - Feltl and Company\n",
      "                </p>\n",
      "<p>\n",
      "                  Sung Ji Nam - Cantor Fitzgerald\n",
      "                </p>\n",
      "<p>\n",
      "                  Thomas Pfister - RedChip\n",
      "                </p>\n",
      "</div>\n",
      "<div class=\"content_part hid\" id=\"article_content\">\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Greetings, and welcome to the Cancer Genetics’ First Quarter 2015 Earnings Conference Call. A brief question-and-answer session will follow the formal presentation. At this time all participants are in a listen-only mode. [Operator Instructions]. As a reminder this conference is being recorded. \n",
      "                </p>\n",
      "<p>\n",
      "                  It’s now my pleasure to introduce your host, Mr. Brian Ritchie [ph] with the Cancer Genetics Investor Relations team. Thank you, Mr. Ritchie. You may begin.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Brian\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you and thank you all for joining us for Cancer Genetics first quarter 2015 earnings conference call. On the call today are company President and Chief Executive Officer, Panna Sharma; and Chief Financial Officer, Ed Sitar.\n",
      "                </p>\n",
      "<p>\n",
      "                  The company issued a news release this morning highlighting the company's financial results and progress on operations. Following the Safe Harbor statement Panna will provide an overview of the first quarter, including recent events and company activity. Ed Sitar will then provide a summary of the quarterly financial results. We will then open the call to questions.\n",
      "                </p>\n",
      "<p>\n",
      "                  I would also like to highlight for everyone on the call or webcast today that accompanying slides are available on the CGI website under the Investor Section that highlights the financial and operating results being discussed on today’s call.\n",
      "                </p>\n",
      "<p>\n",
      "                  I’d like to remind everyone that various remarks about future expectations, plans, and prospects constitute forward-looking statements for purposes of Safe Harbor provisions under the Private Securities Litigation Reform Act of 1995. Cancer Genetics cautions that these forward-looking statements are subject to risks and uncertainties that may cause their actual results to differ materially from those indicated, including risks described in the company’s filings with the SEC. Any forward-looking statements made on this conference call speak only as of today’s date, Thursday, May 12, 2015, and Cancer Genetics does not intend to update any of these forward-looking statements to reflect events or circumstances that occur after today’s date.\n",
      "                </p>\n",
      "<p>\n",
      "                  This conference call is being recorded for audio rebroadcast on Cancer Genetics’ website at www.cancergenetics.com. All participants on this call will be in listen-only mode. The call will be followed by a question-and-answer session.\n",
      "                </p>\n",
      "<p>\n",
      "                  With that I’d like to turn the call over to President and CEO, Panna Sharma for his opening comments. Good morning, Panna?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Panna Sharma\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you Brian. Good morning to all of you today on the call. Thank you for joining us and taking the time to listen and participate in our first quarter 2015 conference call. For 2015, we're off to a very strong start this year and I am pleased to have the opportunity to take you through some of our operational and financial results for the first quarter. \n",
      "                </p>\n",
      "<p>\n",
      "                  The momentum that our business gained in 2014 is continuing to accelerate as we build progress in 2015. The investments that we've made in our genomic portfolio and our acquisitions and our world class collaborations are generating strong results and an increase in our market share and we also making an overall positive impact in the oncology industry and oncology patients. \n",
      "                </p>\n",
      "<p>\n",
      "                  This morning we will update you on our financial results for the first quarter and also on how we're successfully executing on strategic initiatives we laid out on our year end call in March. Following this we'll turn it over to questions and answers. \n",
      "                </p>\n",
      "<p>\n",
      "                  You've heard me say before Cancer Genetics is an emerging leader in providing critical genomic and biomarker information for the personalization of oncology treatment. We offer our proprietary, disease focused and clinical validated genomic tests and services to clinical centers, hospitals, biotechnology and pharmaceutical companies and research organizations. We have a unique and unparalleled global infrastructure for the development and delivery of oncology diagnostics from bench to bed side through our four state-of-the-art facilities in New Jersey, North Carolina, India and China. \n",
      "                </p>\n",
      "<p>\n",
      "                  Our research collaborations with 18 leading academic and research centers, including Mayo Clinic, Memorial Sloan Kettering, The National Cancer Institute, Columbia University, Moffitt Cancer Center and Beth Israel, among many others allows us to access innovation in oncology and genomics, rapidly validates our insights and tests and have unique access to translational oncology programs. We think these collaborations are a true stamp of differentiation and an element of our long term durability as a company. \n",
      "                </p>\n",
      "<p>\n",
      "                  An excellent example of this is our most recent research collaboration with the Moffitt Cancer Center in Tampa, Florida. A series of collaborative studies where diagnostic researcher doctors Heather Jim, Diane Portman, Hailey McCloud and Gillian Bell [ph] will examine a number of genetic variance as predictors for the most common side effects associated with chemotherapy treatment which continues to be one of the most broadly used treatments in cancer. \n",
      "                </p>\n",
      "<p>\n",
      "                  Our second collaboration will examine the world of individual genetic variance and effectiveness of pain control in cancer. Results from this partnership and with other leading cancer research organizations will help guide the development of treatment options and provide critical and validated genomic information that improves patient management and reduces healthcare cost. \n",
      "                </p>\n",
      "<p>\n",
      "                  Beyond impact of research collaborations that will drive future tests and revenue, we have made significant progress in a number of key other areas as we advance our goal of leveraging our market leading molecular oncology capabilities and our global footprint to become the premier partner for personalizing oncology diagnostics from bench to bed side. \n",
      "                </p>\n",
      "<p>\n",
      "                  Earlier I referenced the strong market momentum we continue to see in our business as evidenced by our 206% year-over-year growth and an increase in our contracted revenue with bio pharma customers which is now increased to over 30 million. While Ed will discuss the financial details in greater detail later in this call let me provide you with some key highlights of the trends and results from the first quarter. \n",
      "                </p>\n",
      "<p>\n",
      "                  Importantly our 2014 acquisitions and the synergies we are experiencing from these acquisitions continue to meaningfully contribute to our growth and market penetration. Our new locations in North Carolina and India are driving an increasing number of clinical trials who have been selected to power and provide testing services for, that number’s now well over 70 and we are increasing the number of cross-selling opportunities with some at our largest biopharma customers. \n",
      "                </p>\n",
      "<p>\n",
      "                  In addition, with the additional locations and capabilities we are now running larger size clinical trials, some of which are now spanning multiple laboratory locations. We are very excited about the level of integration we have achieved in the past few quarters and expect this will provide unique value to our customers and to our shareholders. Our infrastructure coupled with the integrated products and portfolio we continue to develop and update and that our customers are rapidly adopting are providing us the backbone to be the leader on oncology diagnostic.\n",
      "                </p>\n",
      "<p>\n",
      "                  Taking a closer look at some of the first quarter financial results, our first quarter year-over-year revenue increased 206% or $3 million to $4.4 million for the three months ended March 31, 2015. This is up sharply from $1.4 million for the three months ending March 31, 2014. Of significance our biopharma service business increased 578% or $2.8 million to $3.2 million in the first quarter of this year and that was up from $491,000 in the same period for 2013. \n",
      "                </p>\n",
      "<p>\n",
      "                  We also improved our gross profit margin significantly year-over-year from 9.8% in 2014 to 28.1% for the first quarter of this year. The majority of this improvement came from improved use of our lab and our staff and better capacity utilization but also improvements to our workflow as we expand our biopharma business and focus our clinical business on more profitable ordering sites. We expect that we will continue to see ongoing gross margin improvements throughout the quarters. \n",
      "                </p>\n",
      "<p>\n",
      "                  A financial metric that is particularly important to note is our overall operating loss which year-over-year changed only 5% while our revenues increased by over 200%. [indiscernible] at this metric because over 90% of our gross profit margin dropped to our operating income. We think this is an important metric and demonstrates both the potential leverage of the business and the rapid progress we have made with the acquired teams.\n",
      "                </p>\n",
      "<p>\n",
      "                  While I am pleased with the momentum and significantly positive trends and results we are seeing in our business I would like to spend a little time discussing some of the key 2015 strategic initiatives and the progress that we are making in a number of important areas. \n",
      "                </p>\n",
      "<p>\n",
      "                  First, with respect to our growing NGS portfolio we are continuing to build a dominant position in the area of hematological malignancies or blood bone cancers with offerings in CRL and myeloid cancer having launched and future launches in multiple myeloma and lymphoid cancer expected for later this year. We have also launched a major driver of growth for us in the India market, our focus Hotspot panel for solid tumors, which targets the five most common solid tumors. \n",
      "                </p>\n",
      "<p>\n",
      "                  Our most recent launch Focus::Myeloid, helps clinicians and cancer centers improve the management and treatment of more than 275,000 patients in the U.S. that are currently living with myeloid malignancies. In addition, over a 100 clinical studies are been conducted by biotech and biopharma companies for compounds targeting myeloid cancers and many of these trials can potentially benefit from CGI’s Focus::Myeloid panel. Focus::Myeloid provides information to improve enrollment, stratification and treatment monitoring for these complex myeloid malignancies.\n",
      "                </p>\n",
      "<p>\n",
      "                  As you know we are actively developing a pipeline of NGS panels targeting a variety of other cancers as well, including hematologic and solid tumors through our internal innovation efforts. In addition, CGI expects to launch NGS panels that are currently in development through Oncospire Genomics, our joint venture with Mayo Clinic. The first of which is multiple myeloma which will be coming out later this year. \n",
      "                </p>\n",
      "<p>\n",
      "                  Our dominant position in hematologic cancer is contributing to our strong growth and contracted revenue expectations from the biopharma and biotech companies. This number increased to $30 million as of the end of the first quarter and in addition we also recently launched an NGS based panel targeting five most common solid tumors for the Indian and broader Asian markets. This NGS panel can be applied to over 370,000 patients in India alone and as a result of this launch we have seen significant uptick in the order volume, revenue and contract discussions with both clinical and research customers.\n",
      "                </p>\n",
      "<p>\n",
      "                  Another key initiative for 2015 is increasing market penetration for our blockbuster genomic technology for HPV-related and surgical cancer testing, FHACT. Last month FHACT received conditional license from --licensure from the New York State Department of Health. We think this is an important milestone since it allows the company now to market the test to women's health centers, gynecologists and oncologists throughout the State of New York, which represents more than 7.6 million women. While cervical cancer screening protocols are relatively well established there remain significant unmet need to supplement current screening methods which are live [ph] and subject to cytological examination or just mapping the presence of the HPV virus. \n",
      "                </p>\n",
      "<p>\n",
      "                  We think that with our test we can not only better detect HPV ovarian cancers earlier but also predict progression to advance disease. CGI's stock test which assesses genomic changes, common [indiscernible] cervical cancer identified women that are at high risk of the disease. In addition, we expect data later this year from a stack-related [ph] study that we just finalized with the National Cancer Institute in over 300 [ph] patients. We also expect a number of publications focused on FHACT and the launch of a multisite health economic study that will demonstrate the economic value and positive patient outcomes related to FHACT. \n",
      "                </p>\n",
      "<p>\n",
      "                  In a critical area of billing and reimbursement, clinical reimbursement we continue to make steady progress. We recently signed an agreement with America's Choice provider network and an independent national network of healthcare providers, which allows us access to over 1,100 out of network payers and allows our diagnostic services to be available to more than 19 million domestic members. I'm very excited about the progress our team has delivered and continues to deliver and where our business is growing today. We believe it's heading -- we really believe it's heading in terms of not only its growth but its impact on patients and the ability to be a true partner for the broader oncology community. \n",
      "                </p>\n",
      "<p>\n",
      "                  We have a broad portfolio of innovative industry leading products with a number of others on the way, a number of key strategic initiatives that will actually broaden our reach and accelerate our growth in both the clinical and biopharma community, I’ll touch on some of those later in the call. And we continue to demonstrate solid market penetration and market adoption, all of which has caused [ph] revenue expansion in our growth going forward. \n",
      "                </p>\n",
      "<p>\n",
      "                  At this point, I'll hand over the call to our CFO, Ed Sitar for a review of our first quarter and our -- first quarter 2015 financial results Ed?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Edward Sitar\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you Panna and good morning everybody. We had a very strong first quarter. Revenues were $4.4 million, a 206% increase over the first quarter of 2014 and 8% higher sequentially than the fourth quarter of 2014. Our revenue is broken down in to three categories; biopharma services or services and test performed as part of clinical trials. Biopharma services revenue grew 578% over the first quarter of 2014. This was driven by organic growth in our Select One business and the synergies from our acquisitions. \n",
      "                </p>\n",
      "<p>\n",
      "                  Clinical services or tests performed for prescribing clinicians. Clinical services revenue decreased by 7% or $66,000 when compared to Q1, 2014. Test mix between Medicare and private insurance companies and the related pricing pressure is responsible for this reduction. To address this challenge the company has adjusted clinical sales efforts to focus on customers that are more profitable and have clear needs for CGI's proprietary portfolio. \n",
      "                </p>\n",
      "<p>\n",
      "                  Discovery services are services provided in the development of new testing [indiscernible] methods for research organizations. Discovery services generated a $166,000 of revenue, up 54% sequentially from the fourth quarter of 2014. Cost of revenues increased to 143% or $1.9 million, cost of revenues from the acquired businesses was $1.2 million. Lab supplies increased by $280,000 and shipping and compensation costs increased by $234,000. \n",
      "                </p>\n",
      "<p>\n",
      "                  Gross margins were 28.1% or $1.2 million, which compares to 9.8% or $140,000 in the first quarter of 2014. The improvement in gross margin percentage is attributable to better utilization of cost along with the margin from our acquired businesses. Total operating expenses were $5.4 million in the quarter compared to $4.1 million in the same period last year. Included in the operating expenses are stock based compensation expenses of $696,000 in Q1, 2015 versus $530,000 in Q1 2014. \n",
      "                </p>\n",
      "<p>\n",
      "                  Research and development expenses in the first quarter increased by 114% or $681,000 to $1.3 million. The increase includes a full quarter of activity from OncoSpire, our joint venture with the Mayo Clinic. In Q1 2015 our share of the loss was $207,000 compared to $12,000 last year. Compensation cost increased by $164,000 and supplies cost increased by $152,000. General and administrative expenses in the first quarter increased 9% or $255,000 to $3 million. Our acquired operations contributed $544,000 to the expense and this was offset by reductions in compensation cost of $327,000.\n",
      "                </p>\n",
      "<p>\n",
      "                  Sales and marketing expenses in the first quarter of 2015 increased 49% or $367,000 to $1.1 million, primarily due to the sales efforts of our acquired operations. Our shares outstanding at March 31, 2015 were 9,831,169. Our diluted shares for the quarter were 9,703,576. Our net loss in the first quarter was $4.3 million or $0.44 per diluted share compared to a loss of $0.27 per diluted share in the corresponding period in 2014.\n",
      "                </p>\n",
      "<p>\n",
      "                  We had total cash at March 31, 2015 of $28.6 million, which includes the $6 million restricted to the company’s loan facility with Wells Fargo. As we discussed in this morning’s release we have successfully refinanced that facility and removed the restriction on that cash. In addition we have secured $4 million of additional borrowing capacity.\n",
      "                </p>\n",
      "<p>\n",
      "                  I will now turn the call back to Panna.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Panna Sharma\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you, Ed. We had a number of value enhancing accomplishments just in the first several months of this year and we expect this to continue driving value for us and our shareholders throughout the rest of 2015. We’ll walk through some of those before we turn to Q&amp;A.\n",
      "                </p>\n",
      "<p>\n",
      "                  First our focus on reimbursement; we’ve been actively working with payers and insurance companies in a very disciplined manner, since revamping our clinical revenue cycle team and we expect to make several updates on coverage and network decisions as the year progresses. Second, additional new developments and commercial partnerships; we’re working on number of commercial partnerships that will increase our proprietary programs, getting in to the marketplace, delivering our suite of services to a greater variety of customers and driving uptick in patient volume. Therefore we think these are very important channel and commercial partnerships that will drive continued growth of our business.\n",
      "                </p>\n",
      "<p>\n",
      "                  Additional product launches is our third, beyond the product launches I touched on earlier we also expect the very first panel through our 50-50 joint venture with the Mayo Clinic, OncoSpire Genomics in multiple myeloma to launch second half of this year. We are on track to developing a very unique panel for the diagnosis and management of multiple myeloma patients. We’re also on track for an accelerated launch of our first targeted test for lung cancer which no other group currently has in the marketplace.\n",
      "                </p>\n",
      "<p>\n",
      "                  We also expect further market penetration for FHACT. The recent conditional licensure from New York State Department of Health is a positive step toward accomplishing this objective. We’ve also retooled ourselves on our clinical sales team to really address this market that has over 2 million tests alone annually that can be improved for the outcome of women. As I said in our March call we’ll continue bringing those women health centers and laboratories into trails and migrating them to routine ordering sites as the year progresses. \n",
      "                </p>\n",
      "<p>\n",
      "                  Fifth, further expanding upon our innovation we see a crucial element to accelerating on our growth. We’ve number of initiatives to make our company more relevant in emerging areas of oncology, namely in neuro-oncology bringing together schematic and germ-line testing for patients in clinical trials, providing testing for hereditary cancers and developing targeted knowledge base in a number of cancers where CGI has unique insight, unique access and knowledge about the disease. \n",
      "                </p>\n",
      "<p>\n",
      "                  We think these four areas will continue to allow us to access a wider place in the market, drive further growth and make us a more valuable partner in delivering oncology diagnostics from bench to bed side. With that I would like to now open the line for questions.\n",
      "                </p>\n",
      "</div>\n",
      "<div class=\"content_part hid\" id=\"article_qanda\">\n",
      "<p>\n",
      "<strong>\n",
      "                    Question-and-Answer Session\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Ladies and gentlemen, we will now be conducting a question-and-answer session. [Operator Instructions]. Our first question comes from the line of Paul Knight with Janney Capital Markets. Please go ahead with your question. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Paul Knight \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Congratulations on the quarter. Regarding the 70 plus clinical trials that you are involved with when would you expect to have color on potential diagnostics or partnerships with pharma on those 70 tests? \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Yes, Paul, first of all thank you, very good question. One of the unique features about our business is exactly is what's you're touching on. Many of the protocols and testing algorithms that we're using in these trails will make their way into the clinic. And when they do make their way into clinic CGI is uniquely positioned to be a partner of choice to provide that testing. I would say today, not all of those will be -- not all of those will develop into companion diagnostics or into testing algorithms. \n",
      "                </p>\n",
      "<p>\n",
      "                  And some of them may result in additional studies with pharma customers. But today the way we look at those 70 we think well over half of those trials are unique. And about half of those trials, about one fourth overall, about 15 to 17, we think have a potential for become a companion testing or therapeutic monitoring type option. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Paul Knight \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  And then where -- how is backlog trending. Obviously I guess it was $25 million to $30 million from Q1 to Q4 and help tone in Q2. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  We continue to see strong momentum in Q2 as well. So it's -- I think Q1 was a good start but we continue to see the same blueprint. And in fact we're actually seeing even more traction with existing customers in Q2. We continue to see the momentum building actually in our biopharma business. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Paul Knight \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Our next question comes from the line of Ben Haynor with Feltl and Company. Please go ahead with your question. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good morning, gentlemen. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good morning Ben.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Edward Sitar\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good morning Ben. How are you?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good, good. Very strong biopharma quarter, was there much next generation sequencing revenue in that or was it kind just starting to get rolling during Q1. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good question Ben, thank you. We -- some of our biggest work in Q1 was in next gen sequencing type. We have a number of trails that have begun adopting our CLL panel, and in fact also myeloid panel. So we don’t break out the NGS specifically as a percentage of revenue but it was a significant driver of growth for us in the quarter. And that's across multiple sites there. Yeah, there is the Hotspot panel that we launched in NDM contributed to it, RCLL [ph] panel that we launched contributed very heavily to it and other NGS study. So it's across all sites, NGS has been a strong driver for us. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Excellent, and then you mentioned the focus Hotspot panel. Do you think that’s something we could see outside of Asia before too long. And then obviously that could do a lot for your cancer specific tracking and various databases, I would assume anyway. Can you give us any color on how the data collection efforts are coming there thus far? \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Yes, so I think we -- so you just asked two questions and I'll try to dice them apart, the first question is really about bringing the tumor Hotspot panel to the U.S. market. We think that's very muted in the U.S. and but there are other providers, not immunologists [ph] to offer something unique in the marketplace. So we think same variation with solid tumor Hotspot panel. In India the timing is very important, number one, to access the market there. The five tumors conditions are the vast majority and are we bringing the right portfolio for the right geography also. So I think we are focusing on Tumor Hotspot but in a way that we can make it unique and value additive to the community there. \n",
      "                </p>\n",
      "<p>\n",
      "                  In terms of database and tracking we’ve started developing targeted databases. As you know the genomic conditions underlying each cancer are slightly different in how they can be interpreted even if it’s the same mutation or aberration can be different in different diseases. So we really are believers in disease specific cancer knowledge basis. So we are in the process of launching not only one for surgical but for CLL and B-cell cancers and also for the tumor Hotspot in India. So those are things that we’ll be providing more color on as the year progresses and we think that will be an important element of the value building for our company.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Great, thanks for taking the questions I will jump back in queue.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Our next question comes from the line of Sung Ji Nam with Cantor Fitzgerald. Please go ahead with your question.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Sung Ji Nam \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Hi, thanks for taking the questions. Was wondering Panna, you talked about sales force [indiscernible] largely last quarter, and was wondering given you had a strong biopharma services, is the sales organization for that part of the business is it pretty much all built out?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  That’s a great question thank you Sung Ji for that. The sales force expansion in biopharma business continues where I would say we are pleased with where it is but the opportunity for us there is very large and growing rapidly. There is a tremendous need for the type of services that we are providing and let me give you a case study that I think will highlight that but head on we are in the early part of our sales force expansion here. So we continue to look for sales people that are the right types of people to help us develop these markets and have the right kind of content and knowledge to carry our brand in to the marketplace. \n",
      "                </p>\n",
      "<p>\n",
      "                  That said we are getting also a lot of synergy from the acquisition and a lot of opportunities from customers of CGI that want the expanded capabilities and customers of the acquisition that want the CGI capabilities. In fact that’s one of the biggest areas that we see today in our pipeline is exactly the synergies that we were hoping for. And so part of that has really, recall kind of almost not necessarily sales driven but really house or capability driven revenue growth which is fantastic, of course but we need to continue investing our sales force and that will drive even further market penetration for us. \n",
      "                </p>\n",
      "<p>\n",
      "                  We also are in discussions with a number of partners and we think that’s very important to know. We are working actively with partners that we think can be massive multiplier effects for our sales force.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Sung Ji Nam \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Great, that’s helpful. And then for the clinical services business do you guys see an increase in the test volume year-over-year and also was the reimbursement pressure pretty much across all the product lines? Thank you.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Yes, I think that’s safe to say, reimbursement pressure was pretty much across the product lines. As you know more in certain areas like FISH and some of the genomic panels than in other areas. We did see slight increase in some of our IHC work that we are doing which is now digital but clearly we did see several points of pressure in the clinical side and we continue to see delays in paying NGS type panels there as well. So I think we’re in for several quarters, broadly not just here but broadly in the clinical community a concern about how a lot of these genomic panels get paid. \n",
      "                </p>\n",
      "<p>\n",
      "                  The good news about our approach that we think is very different, is we really started very active dialog with payers about our targeted disease specific panels. We think this puts us in a driver seat to help them look at how reporting out and the delivery of targeted information that’s already validated for specific diseases helps them. So we think this is very unique position that we are in today versus a lot of our peers out there.\n",
      "                </p>\n",
      "<p>\n",
      "                  The other reason that we think that it’s very important that we are in a better position to get reimbursement is many of the genomic panels are the assay type [ph] test that are offered by our peers. They don’t have the comprehensive lab capabilities, often times insurers and third party groups will not pay for what perceive to be as the high end of the test, especially if it’s not already in the guideline unless you already have done a test using traditional technology or methodologies that are widely accepted, regardless of how accurate or how specific they are. \n",
      "                </p>\n",
      "<p>\n",
      "                  For example, often times in a blood-borne cancer people rely initially on a flow-based test and then reflex to a genomic panel or they rely on a FISH-based test that may or may not give accurate results and then go on to a more thorough comprehensive genomic panel. In both those cases we’re in a great position because we can reflux right now in our lab and we can demonstrate through the clinical history of that test, that the test A was a clear suggestion to then do Test B. That really helps us significantly and what we’re finding from feedback from payers that our new Head of Vendors [ph] that Dr. Randy Goodman [ph] has been leading is that comprehensive capability of having the entire disease focused library at one site is a massive win for us. \n",
      "                </p>\n",
      "<p>\n",
      "                  And we have number of case studies now that we’ve developed not only for our sales force but for the insurers to show them for example in CLL or myeloid cancer or in cervical cancer how having the complete capabilities and then being able to put our cash into the whole stream, helps to make a decision, drives cost down and gets people to treatment faster and without having to resample the patient. So we think that industry is maturing in terms of its approach to listening about these issues. But we’re still think it’s going to be a quarter or two before, they say okay we will routinely pay for all the stuff, day in and day out. \n",
      "                </p>\n",
      "<p>\n",
      "                  And as you know these days it’s a very, very fractured industry out there in terms of how the insurance groups pay for it, but I think we’re in a great position and we got great data and of course our business model where we can do all the testing helps us.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator \n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. [Operator Instructions]. Our next question comes from the line of Thomas Pfister with RedChip. Please go ahead with your question.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Thomas Pfister \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good morning Paul and good morning Ed. How are you guys \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good yourself?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Thomas Pfister \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Good, good. My first question here, I think you mentioned earlier that you are shifting your sales strategy whatever with the customer you’re targeting in your Clinical Services segment. Can you just please give some color on that shift?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Sure. I think as I can elaborate our focus is ready go on more of a profitable customer segments and customers who are going to be able to bring entire portfolio capabilities too. So I think just a quality of revenue, quality of process as you’re seeing we’ve have massive growth and we continue to in our Biopharma business, and as we continue to grow the Biopharma business we have got to put the resources at the right places and look at profitability, for cash profitability per count, potential profitability on the same measures. \n",
      "                </p>\n",
      "<p>\n",
      "                  So we’ve really decided to focus slightly more in Biopharma in terms of better sales model and then focusing on the clinical sites that we think are going to big long-term profit drivers for us and our shareholders.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Thomas Pfister \n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Okay, great. Thanks for the color there and just the follow-up question for me. Can you please give some insight in maybe the market opportunity you would see with our Pharmacogenomics panel and kind of what type of competition do you see for that panel out there right now in the marketplace?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  So market opportunity if I can assume there we’re mostly focusing Pharmacogenomics for oncology and that we do think that can be spilled over into other disease stage as well. Many of the new targeted drugs and many of the new combination regimens do require some Pharmacogenomic profiling or Pharmacogenomic monitoring and so we think we are very uniquely positioned to deliver that information because we’re already delivering information on the schematic genome.\n",
      "                </p>\n",
      "<p>\n",
      "                  Again that was one of the driver kind of our desire to do the acquisition of Gentris last year. In terms of market size and market opportunity we haven’t specifically tied PGX in oncology, that PGX itself in the U.S. is close to a $1 billion opportunity but PGX also is being done [ph] not only in patients but also for clinical trials, naturally the majority of our work is being done today. A lot of that will now go into the clinical settings as well as we get unique data and unique insight.\n",
      "                </p>\n",
      "<p>\n",
      "                  And we think this for us add potentially tens of thousands of new tests for us every quarter that we can have access too. And then more importantly we think more value towards clinical sites because we’ll have the data not only of the disease and cancer and also have the data on the pharmacogenomic profile of that patient, and marrying those together we think is a big motive differentiation. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Thomas Pfister \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Okay, great. Thanks so much for the color there. And thank you for taking my questions. I will hop back in the queue. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Our next question is a follow-up from the line of Ben Haynor with Feltl and Company. Please go ahead with your follow-up.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thanks for taking my follow-up. Yeah, just kind of following on the last question, not too long we saw that Johns Hopkins Kimmel Cancer study [indiscernible] science translational medicine. It looks like you guys had some poor sight in looking at Gentris and making that acquisition. Have you seen quite a bit of additional interest from biopharma as a result of the publication of that study or still too early days, I know you guys have somewhat unique capabilities with CGI and Gentris testing combined. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  I think it is a great question. In terms of translation of that opportunity, of that issue into [indiscernible] I think it is early but it’s beginning. I think almost every conversation that I have with academic research groups about lack of adequate -- line information and the profile of patients we have had. I was at Mayo last week, to discuss that in length, we discussed in terms of how it will impact, how we do variant cause and how we potentially can gather germ line [ph] to our observed germ line materials at the same time, it’s impacting the entire industry. I think we can definitely not stick our head in the sand say we are going to come with the bio-informatic driven solution. And we have to aware that germ line and schematics, remove the tumor from the DNA -- remove the DNA from the tumor itself is very important and it is something that our team has brought up several years back and that had let us to have vision of bringing the two pieces together. \n",
      "                </p>\n",
      "<p>\n",
      "                  So I don’t think it’s -- I think there is a lot of thought and lot of discussion, I think mostly those pharma partners that we have talked to, say the same thing and they do recognize that as an issue and as a way that can potentially change their impact. Now for us it’s not only having the capabilities of doing both and having data based capabilities but also for us because we have targeted panels for disease specific we actually tune these panels to be more focused on that disease category so changes for us reporting out germ line variance is minimized. And that’s also an important -- what I think about it is truly a knowledge base driven approach. \n",
      "                </p>\n",
      "<p>\n",
      "                  That said the insurance companies pay for that. They understand it, patients and clinicians who can understand know it takes some time till it makes it’s way well into wider marketplace but it is definitely a topic that is relevant today and probably drives significant more opportunity for us and biopharma earlier and then into the clinical setting after it.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Ben Haynor \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Great, that is very helpful. That is all I had gentlemen, thank you very much. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you, Ben.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Our next question is a follow-up from the line of Paul Knight with Janney Capital Markets. Please go ahead with your follow-up.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Paul Knight \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Hi, good morning. Could you give us a quick recap of kind of the major tests that you are expecting in the rest of the year?\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Yes, absolutely so we expect to a launch a lymphoid cancer panel and it is very unique and we also expect to launch a multiple myeloma panel, that would be through a joint venture with Mayo Clinic and we also expect to -- the first version of the lung cancer test in Mayo Clinic into market as well. We also expect to then have our PGX or pharmacogenomics comprehensive profiler into the market later this year and those are the four that we think will be come on track for the second-half of this year.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"question\">\n",
      "                      Paul Knight \n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Okay, thank you.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "<span class=\"answer\">\n",
      "                      Panna Sharma\n",
      "                    </span>\n",
      "</strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you very much. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. [Operator Instructions]. Ladies and gentlemen, there are no further questions at this time. I would now like to turn the floor back over to management for closing remarks.\n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Panna Sharma\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you. Great questions. Again at Cancer Genetics our pursuit is to build a unique company that is able to change the paradigm in oncology diagnostics and leverage our insight and capabilities to be a true partner from bench to bed side as we change patient outcomes and improve drug trials in oncology. To that I think we’ve made a number of very important milestones, both operationally as well as from a portfolio over the last several months and we continue to see very strong momentum in our business. We’ve a number of milestones that we continue to see through the remainder of this year that will drive even accelerated top line growth potentially continue to drive margin improvement and drive more market share for us.\n",
      "                </p>\n",
      "<p>\n",
      "                  So again thank you for following our story and I look forward to visiting with many of you in the future. \n",
      "                </p>\n",
      "<p>\n",
      "<strong>\n",
      "                    Operator\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  Thank you ladies and gentlemen, this does conclude our teleconference for today. You may now disconnect your lines at this time. Thank you for your participation and have a wonderful day. \n",
      "                </p>\n",
      "</div>\n",
      "<div class=\"content_part hid\" id=\"article_disclaimer\">\n",
      "<p>\n",
      "<strong>\n",
      "                    Copyright policy: \n",
      "                  </strong>\n",
      "                  All transcripts on this site are the copyright of Seeking Alpha. However, we view them as an important resource for bloggers and journalists, and are excited to contribute to the democratization of financial information on the Internet. (Until now investors have had to pay thousands of dollars in subscription fees for transcripts.) So our reproduction policy is as follows: \n",
      "                  <strong>\n",
      "                    You may quote up to 400 words of any transcript on the condition that you attribute the transcript to Seeking Alpha and either link to the original transcript or to www.SeekingAlpha.com. All other use is prohibited.\n",
      "                  </strong>\n",
      "</p>\n",
      "<p>\n",
      "                  THE INFORMATION CONTAINED HERE IS A TEXTUAL REPRESENTATION OF THE APPLICABLE COMPANY'S CONFERENCE CALL, CONFERENCE PRESENTATION OR OTHER AUDIO PRESENTATION, AND WHILE EFFORTS ARE MADE TO PROVIDE AN ACCURATE TRANSCRIPTION, THERE MAY BE MATERIAL ERRORS, OMISSIONS, OR INACCURACIES IN THE REPORTING OF THE SUBSTANCE OF THE AUDIO PRESENTATIONS. IN NO WAY DOES SEEKING ALPHA ASSUME ANY RESPONSIBILITY FOR ANY INVESTMENT OR OTHER DECISIONS MADE BASED UPON THE INFORMATION PROVIDED ON THIS WEB SITE OR IN ANY TRANSCRIPT. USERS ARE ADVISED TO REVIEW THE APPLICABLE COMPANY'S AUDIO PRESENTATION ITSELF AND THE APPLICABLE COMPANY'S SEC FILINGS BEFORE MAKING ANY INVESTMENT OR OTHER DECISIONS.\n",
      "                </p>\n",
      "<p>\n",
      "                  If you have any additional questions about our online transcripts, please contact us at: \n",
      "                  <a href=\"mailto:transcripts@seekingalpha.com\" rel=\"nofollow\">\n",
      "<u>\n",
      "                      transcripts@seekingalpha.com\n",
      "                    </u>\n",
      "</a>\n",
      "                  . Thank you!\n",
      "                </p>\n",
      "</div>\n",
      "<div id=\"article_slice\">\n",
      "</div>\n",
      "<div class=\"cleaner\">\n",
      "</div>\n",
      "<div id=\"article_source\">\n",
      "                Source: \n",
      "                <a href=\"/article/3174886-cancer-genetics-cgix-ceo-panna-sharma-on-q1-2015-results-earnings-call-transcript?source=cc\">\n",
      "                  Cancer Genetics' (CGIX) CEO Panna Sharma on Q1 2015 Results - Earnings Call Transcript\n",
      "                </a>\n",
      "</div>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "with open(collection + '/' + test_file) as inf:\n",
    "    txt = inf.read()\n",
    "    soup_out = soup(txt, \"html.parser\")\n",
    "print(soup_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# for h4 in soup_out.findAll('strong'):\n",
    "#     print(h4.text)\n",
    "temp = soup_out.find_all('span',{'class':'question'})\n",
    "# for element in temp:\n",
    "#     out =(element.attrs[\"content\"])  \n",
    "#     print(out)\n",
    "# print(temp)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_temp = temp[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'\\n                      Paul Knight \\n                    ']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_temp.contents"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract Firm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cancer Genetics, Inc.\n"
     ]
    }
   ],
   "source": [
    "temp = soup_out.find(id='article_participants')\n",
    "for element in temp.findAll('a'):\n",
    "    if element.get('title') != None :\n",
    "        print(element.get('title'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def extract_firm(input_soup):\n",
    "    temp = soup_out.find(id='article_participants')\n",
    "    for element in temp.findAll('a'):\n",
    "        if element.get('title') != None :\n",
    "            return(str(element.get('title')))   \n",
    "    return None"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract Quarter and Year"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "                  Q1 2015 Earnings Conference Call\n",
      "                \n",
      "('quarter: ', 'Q1')\n",
      "('year: ', '2015')\n"
     ]
    }
   ],
   "source": [
    "temp = soup_out.find(id='article_participants')\n",
    "for element in temp.findAll('p'):\n",
    "    if(re.search('call',element.contents[0],flags=re.IGNORECASE)):\n",
    "        print(element.contents[0])\n",
    "        call_title = element.contents[0].split()\n",
    "        print('quarter: ', str(call_title[0]))\n",
    "        print('year: ', str(call_title[1]))        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def extract_time(input_soup):\n",
    "    temp = input_soup.find(id='article_participants')\n",
    "    for element in temp.findAll('p'):\n",
    "        if(re.search('call',element.contents[0],flags=re.IGNORECASE)):\n",
    "            call_title = element.contents[0].split()\n",
    "            return str(call_title[0]),str(call_title[1])\n",
    "    return None,None"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Verify if Analyst Present"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyst Present\n"
     ]
    }
   ],
   "source": [
    "temp = soup_out.find(id='article_participants')\n",
    "for element in temp.findAll('strong'):\n",
    "    if(re.search('analyst',element.contents[0],flags=re.IGNORECASE)):\n",
    "        print('Analyst Present')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def test_analyst(input_soup):\n",
    "    temp = input_soup.find(id='article_participants')\n",
    "    for element in temp.findAll('strong'):\n",
    "        if(re.search('analyst',element.contents[0],flags=re.IGNORECASE)):\n",
    "            return(True)\n",
    "    return(None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_analyst(soup_out)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get Analyst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(u'Analysts', '###')\n",
      "\n",
      "                  Janet Kloppenburg - JJK Research\n",
      "                \n",
      "\n",
      "                  Kimberly Greenberger - Morgan Stanley\n",
      "                \n",
      "\n",
      "                  Gene Vladimirov - Nomura Securities\n",
      "                \n",
      "\n",
      "                  Lorraine Hutchinson - Bank of America\n",
      "                \n",
      "\n",
      "                  Matthew McClintock - Barclays Capital\n",
      "                \n",
      "\n",
      "                  Courtney Wilson - Cowen and Company\n",
      "                \n",
      "\n",
      "                  Jennifer Black - Jennifer Black & Associates\n",
      "                \n",
      "\n",
      "                  Neely Tamminga - Piper Jaffray\n",
      "                \n",
      "\n",
      "                  Thomas Filandro - Susquehanna International\n",
      "                \n",
      "\n",
      "                  Jennifer Davis - Buckingham Research Group\n",
      "                \n",
      "\n",
      "                  Marni Shapiro - The Retail Tracker\n",
      "                \n",
      "\n",
      "                  Anna Andreeva - Oppenheimer\n",
      "                \n",
      "\n",
      "                  Susan Anderson - FBR Capital Markets\n",
      "                \n",
      "\n",
      "                  Lindsay Drucker-Mann - Goldman Sachs\n",
      "                \n",
      "\n",
      "                  Dorothy Lakner - Topeka Capital Markets\n",
      "                \n",
      "\n",
      "                  Christian Buss - Credit Suisse\n",
      "                \n",
      "\n",
      "                  Rebecca Duval - BlueFin Research Partners\n",
      "                \n",
      "\n",
      "\n",
      "FOUND\n"
     ]
    }
   ],
   "source": [
    "temp = soup_out.find(id='article_participants')\n",
    "# print(soup_out)\n",
    "print_flag = False\n",
    "# print(temp) \n",
    "\n",
    "# for element in temp.find_all('p'):\n",
    "# #     break\n",
    "#     if(re.search('analyst',element.text.strip(),flags=re.IGNORECASE)):\n",
    "#         print(element.text.strip())\n",
    "# #         print(element.next_element().text.strip())\n",
    "#         element.next_elements\n",
    "#         print(element)\n",
    "#         out_test = element.next_element\n",
    "#         print(out_test)     \n",
    "#         out_test = out_test.next_element\n",
    "#         print(out_test)             \n",
    "\n",
    "\n",
    "for element in temp.find_all('p'):\n",
    "#     break\n",
    "    if(re.search('analyst',element.text,flags=re.IGNORECASE)):\n",
    "        print(element.text.strip(),'###')\n",
    "        element = element.find_next()\n",
    "        while  (1):\n",
    "            element = element.find_next()\n",
    "            print(element.contents[0])\n",
    "#             print(element.text.strip())\n",
    "            if re.search('strong',unicode(element),flags=re.IGNORECASE) or \\\n",
    "            re.search('div',unicode(element),flags=re.IGNORECASE):\n",
    "                print('FOUND')\n",
    "                break\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(u'Executives', '###')\n",
      "Brian Logan - VP, IR and Controller\n",
      "Michael Jeffries - CEO\n",
      "Joanne Crevoiserat - CFO and EVP of Finance\n",
      "Jonathan Ramsden - COO\n",
      "Analysts\n"
     ]
    }
   ],
   "source": [
    "temp = soup_out.find(id='article_participants')\n",
    "# print(soup_out)\n",
    "print_flag = False\n",
    "# print(temp) \n",
    "\n",
    "# for element in temp.find_all('p'):\n",
    "# #     break\n",
    "#     if(re.search('analyst',element.text.strip(),flags=re.IGNORECASE)):\n",
    "#         print(element.text.strip())\n",
    "# #         print(element.next_element().text.strip())\n",
    "#         element.next_elements\n",
    "#         print(element)\n",
    "#         out_test = element.next_element\n",
    "#         print(out_test)     \n",
    "#         out_test = out_test.next_element\n",
    "#         print(out_test)             \n",
    "\n",
    "\n",
    "for element in temp.find_all('p'):\n",
    "    if(re.search('executive',element.text,flags=re.IGNORECASE)):\n",
    "        print(element.text.strip(),'###')\n",
    "        element = element.find_next()\n",
    "        while  (1):\n",
    "            element = element.find_next()\n",
    "#             print(element.contents[0])\n",
    "            print(element.text.strip())\n",
    "            if re.search('strong',unicode(element),flags=re.IGNORECASE) or \\\n",
    "            re.search('div',unicode(element),flags=re.IGNORECASE):\n",
    "#                 print('FOUND')\n",
    "                break\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'ResultSet' object has no attribute 'next_siblings'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-127-66f1017624f9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mlabel\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m     \u001b[0;31m# extract all next text nodes before the next \"strong\" element or the end\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m     \u001b[0mvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mtext\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtakewhile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnot_tag\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp_s\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_siblings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m     \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'ResultSet' object has no attribute 'next_siblings'"
     ]
    }
   ],
   "source": [
    "from itertools import takewhile\n",
    "from bs4 import BeautifulSoup, Tag\n",
    "\n",
    "labels = temp.find_all(\"strong\")\n",
    "p_s = temp.find_all(\"strong\")\n",
    "# print(labels)\n",
    "# # first strong element is a Executive name\n",
    "business_name = labels[0].get_text()\n",
    "\n",
    "not_tag = lambda elm: not isinstance(elm, Tag)\n",
    "\n",
    "# # going over all the strong \"labels\"\n",
    "for label in labels:\n",
    "    # extract all next text nodes before the next \"strong\" element or the end\n",
    "    value = \" \".join([text.strip() for text in takewhile(not_tag, label.next_siblings)])\n",
    "\n",
    "    print(label.get_text(strip=True), value)      "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Iterate through all the files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FILE\tANALYST_Flag\tFIRM\tQuarter\tYear\n",
      "1.html\tNone\tActions Semiconductor Co., Ltd.\tQ1\t2015\n",
      "10.html\tTrue\tCancer Genetics, Inc.\tQ1\t2015\n",
      "100.html\tTrue\tStartek Inc.\tQ1\t2015\n",
      "1000.html\tTrue\tAlibaba Group Holding Limited\tQ4\t2015\n",
      "10000.html\tTrue\tAbercrombie & Fitch\tQ3\t2014\n"
     ]
    }
   ],
   "source": [
    "collection = \"html\"\n",
    "\n",
    "n_test=5\n",
    "file_list = os.listdir(collection)\n",
    "out_list =[]\n",
    "# for i in range(len(file_list)):\n",
    "print('FILE\\tANALYST_Flag\\tFIRM\\tQuarter\\tYear') \n",
    "\n",
    "for i in range(n_test):\n",
    "    with open(collection + '/' + file_list[i]) as inf:\n",
    "        out_list.append(file_list[i])\n",
    "        txt = inf.read()\n",
    "        soup_out = soup(txt, \"html.parser\")\n",
    "        out_firm = extract_firm(soup_out)\n",
    "        out_time = extract_time(soup_out)\n",
    "        analyst_flag = test_analyst(soup_out)\n",
    "        print('%s\\t%s\\t%s\\t%s\\t%s' ) %( file_list[i],analyst_flag,out_firm, out_time[0],out_time[1])\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_pd = pd.DataFrame(out_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_pd.columns=['file']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10000.html</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         file\n",
       "0      1.html\n",
       "1     10.html\n",
       "2    100.html\n",
       "3   1000.html\n",
       "4  10000.html"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "out_pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-eb4deaa4cf13>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miterrows\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'ifor'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mifor_val\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "for i, row in df.iterrows():\n",
    "    df.set_value(i,'ifor',ifor_val)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  },
  "widgets": {
   "state": {},
   "version": "1.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
